docs/api-reference/next.config.js/disabling-http-keep-alive.md

```diff
@@ -4,19 +4,9 @@ description: Next.js will automatically use HTTP Keep-Alive by default. Learn mo
 
 # Disabling HTTP Keep-Alive
 
-Next.js automatically polyfills [node-fetch](/docs/basic-features/supported-browsers-features#polyfills) and enables [HTTP Keep-Alive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive) by default. You may want to disable HTTP Keep-Alive for certain `fetch()` calls or globally.
+Next.js automatically polyfills `fetch()` in the Node.js environment with [node-fetch](/docs/basic-features/supported-browsers-features#polyfills) and enables [HTTP Keep-Alive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive) by default.
 
-For a single `fetch()` call, you can add the agent option:
-
-```js
-import { Agent } from 'https'
-
-const url = 'https://example.com'
-const agent = new Agent({ keepAlive: false })
-fetch(url, { agent })
-```
-
-To override all `fetch()` calls globally, you can use `next.config.js`:
+To disable HTTP Keep-Alive for all `fetch()` calls on the server-side, open `next.config.js` and add the `httpAgentOptions` config:
 
 ```js
 module.exports = {
```

Comment:
I believe this uses the same Agent behavior so it probably wont work either on Node.js 18

cc @Ethan-Arrowood who implemented undici and might know more

Reply:
Based on our testing, unfortunately these agent options aren't compatible with undici and there's no equivalent alternative.

We had to intentionally disable Node 18's undici with `--no-experimental-fetch`

Reply:
undici's fetch accepts [dispatcher](https://github.com/nodejs/undici#undicifetchinput-init-promise) which fulfills a similar role as `agent` though it requires undici specific Pool/Agent/etc instead of node's http agent.